build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoCommand.java

```diff
@@ -0,0 +1,107 @@
+package org.jfrog.build.extractor.go.extractor;
+
+import org.apache.commons.lang.StringUtils;
+import org.jfrog.build.api.Artifact;
+import org.jfrog.build.api.Build;
+import org.jfrog.build.api.Dependency;
+import org.jfrog.build.api.Module;
+import org.jfrog.build.api.builder.ModuleBuilder;
+import org.jfrog.build.api.util.Log;
+import org.jfrog.build.client.ArtifactoryVersion;
+import org.jfrog.build.extractor.clientConfiguration.ArtifactoryBuildInfoClientBuilder;
+import org.jfrog.build.extractor.clientConfiguration.ArtifactoryClientBuilderBase;
+import org.jfrog.build.extractor.clientConfiguration.client.ArtifactoryBaseClient;
+import org.jfrog.build.util.VersionCompatibilityType;
+import org.jfrog.build.util.VersionException;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+/**
+ * Base class for go build and go publish commands.
+ */
+abstract class GoCommand implements Serializable {
+
+    protected static final String SHA1 = "SHA1";
+    protected static final String MD5 = "MD5";
+    protected static final String LOCAL_GO_MOD_FILENAME = "go.mod";
+    private static final long serialVersionUID = 1L;
+    private static final ArtifactoryVersion MIN_SUPPORTED_ARTIFACTORY_VERSION = new ArtifactoryVersion("6.10.0");
+    static final Pattern moduleNameRegexPattern = Pattern.compile("module \"?([\\w\\.@:%_\\+-.~#?&]+/?.+\\w)");
+
+    ArtifactoryClientBuilderBase clientBuilder;
+    Path path;
+    String moduleName;
+    Log logger;
+
+    GoCommand(ArtifactoryBuildInfoClientBuilder clientBuilder, Path path, Log logger) throws IOException {
+        this.clientBuilder = clientBuilder;
+        this.logger = logger;
+        this.path = path;
+        parseModuleName();
+    }
+
+    private void parseModuleName() throws IOException {
+        Path modPath = Paths.get(path.toString(), LOCAL_GO_MOD_FILENAME);
+        Stream<String> stream = Files.lines(modPath, StandardCharsets.UTF_8);
+        stream.forEach(line -> {
+            if (moduleNameRegexPattern.matcher(line).matches()) {
+                moduleName = line;
+            }
+        });
+        moduleName = StringUtils.split(moduleName, " ")[1];
+    }
+
+    protected void preparePrerequisites(String repo, ArtifactoryBaseClient client) throws VersionException, IOException {
+        validateArtifactoryVersion(client);
+        validateRepoExists(repo, client,"Target repo must be specified");
```

Comment:
Shouldn't it be "Source repo must be specified" when running go-build?

Reply:
The repo existence is checked in run and publish. I changed the error msg to be more general.